controllers/replication-controller/persistentvolumeclaim_controller.go

```diff
@@ -144,6 +147,32 @@ func (r *PersistentVolumeClaimReconciler) Reconcile(ctx context.Context, req ctr
 	remoteClaim := &v1.PersistentVolumeClaim{}
 	remotePVCName := ""
 	remotePVCNamespace := ""
+	if remoteClusterID != controller.Self && r.AllowPVCCreationOnTarget {
+		// if its not single cluster then check the pv status and create pvc on target cluster
+		if remotePV.Status.Phase == v1.VolumeAvailable && remotePV.Spec.ClaimRef != nil {
+			remoteClaim.Spec.AccessModes = remotePV.Spec.AccessModes
+			remoteClaim.Spec.Resources.Requests = v1.ResourceList{
+				v1.ResourceStorage: remotePV.Spec.Capacity[v1.ResourceStorage],
+			}
+			// updating pvc annotations
+			updatePVCAnnotations(remoteClaim, remoteClusterID, remotePV)
+			// Adding finalizer to pvc
+			remoteClaim.Finalizers = []string{controller.ReplicationFinalizer}
```

Comment:
Do we need this finalizer? Do we handle removing this finalizer from the object? If not, the user need to remove this finalizer manually for every PVC object created for the replicated volumes. Please make a note to update the documentation of this is the expected behavior.